src/test/java/com/akademia/educators_back/service/impl/CommentServiceImplTestVerify.java

```diff
@@ -0,0 +1,179 @@
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+public class CommentServiceImplTestVerify {
+
+    @Mock
+    private CommentRepository commentRepository;
+    @Mock
+    private CommentMapper commentMapper;
+    @Mock
+    private CommentValidator commentValidator;
+
+    @InjectMocks
+    private CommentServiceImpl commentService;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void verifyMethodAddComment() {
+        CommentTo commentTo = CommentTo.builder()
+                .id(1L)
+                .answer("Sample answer")
+                .userId(123L)
+                .problemToId(456L)
+                .build();
+
+        CommentEntity commentEntity = CommentEntity.builder()
+                .id(1L)
+                .answer("Sample answer")
+                .userEntity(Mockito.mock(UserEntity.class))
+                .problemEntity(Mockito.mock(ProblemEntity.class))
+                .build();
+
+
+        when(commentMapper.toCommentEntity(commentTo)).thenReturn(commentEntity);
+        doNothing().when(commentValidator).validation(commentTo);
+
+        commentService.addComment(commentTo);
+
+        verify(commentValidator, times(1)).validation(commentTo);
+        verify(commentMapper, times(1)).toCommentEntity(commentTo);
+        verify(commentRepository, times(1)).save(commentEntity);
+    }
+
+    @Test
+    public void verifyMethodDeleteComment() {
+        CommentTo commentTo = CommentTo.builder()
+                .id(1L)
+                .answer("Sample answer")
+                .userId(123L)
+                .problemToId(456L)
+                .build();
+
+        CommentEntity commentEntity = CommentEntity.builder()
+                .id(1L)
+                .answer("Sample answer")
+                .userEntity(Mockito.mock(UserEntity.class))
+                .problemEntity(Mockito.mock(ProblemEntity.class))
+                .build();
+
+        when(commentMapper.toCommentEntity(commentTo)).thenReturn(commentEntity);
+        doNothing().when(commentValidator).validation(commentTo);
+
+        commentService.deleteComment(commentTo);
+
+        verify(commentValidator, times(1)).validation(commentTo);
+        verify(commentRepository, times(1)).delete(commentEntity);
+    }
+
+    @Test
+    public void verifyMethodUpdateComment() {
+        Long commentId = 1L;
+        String updatedAnswer = "Updated answer";
+
+        CommentTo updatedCommentTo = CommentTo.builder()
```

Comment:
Jak używamy tych samych danych to wyrzucić je do prywatnej metody na końcu lub klasy z danymi testowymi